pkg/cmd/server/start/master_args.go

```diff
@@ -254,6 +254,7 @@ func (args MasterArgs) BuildSerializeableMasterConfig() (*configapi.MasterConfig
 		if builtInKubernetes {
 			config.KubeletClientInfo.CA = admin.DefaultRootCAFile(args.ConfigDir.Value())
 			config.KubeletClientInfo.ClientCert = kubeletClientInfo.CertLocation
+			config.ServiceAccountConfig.MasterCA = admin.DefaultRootCAFile(args.ConfigDir.Value())
```

Comment:
you're right. still feels like it should be in the section with the service account settings, but you're right, it still needs to be guarded by `if args.ListenArg.UseTLS()`